### ISSUE TITLE
[BUG] panic: "invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by ialidzhikov** [gardener/etcd-druid#44]
**Describe the bug:**
etcd-druid panics with nil pointer.

**Expected behavior:**

**How To Reproduce (as minimally and precisely as possible):**

**Logs:**

```
E0416 09:08:10.455443       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 317 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x14e9740, 0x23fb600)
	/go/src/github.com/gardener/etcd-druid/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/gardener/etcd-druid/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x14e9740, 0x23fb600)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/gardener/etcd-druid/controllers.(*EtcdReconciler).getMapFromEtcd(0xc0000d4050, 0xc00086c500, 0x3fb999999999999a, 0x4, 0x0)
	/go/src/github.com/gardener/etcd-druid/controllers/etcd_controller.go:881 +0x1276
github.com/gardener/etcd-druid/controllers.(*EtcdReconciler).reconcileEtcd(0xc0000d4050, 0xc00086c500, 0xc00086c500, 0x0, 0x0, 0x0)
	/go/src/github.com/gardener/etcd-druid/controllers/etcd_controller.go:724 +0x4d
github.com/gardener/etcd-druid/controllers.(*EtcdReconciler).reconcile(0xc0000d4050, 0x18da080, 0xc000048248, 0xc00086c500, 0xc000845c40, 0x2, 0x2, 0x18a9ec0)
	/go/src/github.com/gardener/etcd-druid/controllers/etcd_controller.go:227 +0x27c
github.com/gardener/etcd-druid/controllers.(*EtcdReconciler).Reconcile(0xc0000d4050, 0xc000187300, 0x16, 0xc0005c4d00, 0xb, 0xc000758c00, 0x1, 0xc000758cc8, 0x478588)
	/go/src/github.com/gardener/etcd-druid/controllers/etcd_controller.go:189 +0x30b
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000184840, 0x15400e0, 0xc00061a400, 0x43eb00)
	/go/src/github.com/gardener/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000184840, 0x0)
	/go/src/github.com/gardener/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc000184840)
	/go/src/github.com/gardener/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00069c530)
	/go/src/github.com/gardener/etcd-druid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00069c530, 0x3b9aca00, 0x0, 0x1, 0xc00015a0c0)
	/go/src/github.com/gardener/etcd-druid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc00069c530, 0x3b9aca00, 0xc00015a0c0)
	/go/src/github.com/gardener/etcd-druid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/src/github.com/gardener/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193 +0x328
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1347ff6]

goroutine 317 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/gardener/etcd-druid/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x14e9740, 0x23fb600)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/gardener/etcd-druid/controllers.(*EtcdReconciler).getMapFromEtcd(0xc0000d4050, 0xc00086c500, 0x3fb999999999999a, 0x4, 0x0)
	/go/src/github.com/gardener/etcd-druid/controllers/etcd_controller.go:881 +0x1276
github.com/gardener/etcd-druid/controllers.(*EtcdReconciler).reconcileEtcd(0xc0000d4050, 0xc00086c500, 0xc00086c500, 0x0, 0x0, 0x0)
	/go/src/github.com/gardener/etcd-druid/controllers/etcd_controller.go:724 +0x4d
github.com/gardener/etcd-druid/controllers.(*EtcdReconciler).reconcile(0xc0000d4050, 0x18da080, 0xc000048248, 0xc00086c500, 0xc000845c40, 0x2, 0x2, 0x18a9ec0)
	/go/src/github.com/gardener/etcd-druid/controllers/etcd_controller.go:227 +0x27c
github.com/gardener/etcd-druid/controllers.(*EtcdReconciler).Reconcile(0xc0000d4050, 0xc000187300, 0x16, 0xc0005c4d00, 0xb, 0xc000758c00, 0x1, 0xc000758cc8, 0x478588)
	/go/src/github.com/gardener/etcd-druid/controllers/etcd_controller.go:189 +0x30b
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000184840, 0x15400e0, 0xc00061a400, 0x43eb00)
	/go/src/github.com/gardener/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000184840, 0x0)
	/go/src/github.com/gardener/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc000184840)
	/go/src/github.com/gardener/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00069c530)
	/go/src/github.com/gardener/etcd-druid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00069c530, 0x3b9aca00, 0x0, 0x1, 0xc00015a0c0)
	/go/src/github.com/gardener/etcd-druid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc00069c530, 0x3b9aca00, 0xc00015a0c0)
	/go/src/github.com/gardener/etcd-druid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/src/github.com/gardener/etcd-druid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193 +0x328
```

**Screenshots (if applicable):**

**Environment (please complete the following information):**
 - Etcd version/commit ID :
 - Etcd-druid version/commit ID :
 - Cloud Provider [All/AWS/GCS/ABS/Swift/OSS]:

**Anything else we need to know?**:
etcd-druid: v0.1.14

**Comment by georgekuruvillak** [comment]
fixed in this #42

**Comment by ialidzhikov** [comment]
Fixed with [v0.1.15](https://github.com/gardener/etcd-druid/releases/tag/v0.1.15) release.
